source/agora/consensus/Ledger.d

```diff
@@ -1489,9 +1516,10 @@ public class ValidatingLedger : Ledger
     {
         const next_height = this.getBlockHeight() + 1;
 
-        data.enrolls = this.getCandidateEnrollments(next_height);
-        data.missing_validators = this.getCandidateMissingValidators(next_height);
-        data.tx_set = this.getCandidateTransactions(next_height, max_txs);
+        auto utxo_finder = this.utxo_set.getUTXOFinder();
```

Comment:
by using the same utxo_finder with double spend protection, we simply force a UTXO to be referenced only once in a block, either in a enrollment or a TX.